### DISCUSSION TITLE
Monitor Z2M with icinga/nagios (or other solution)

### DISCUSSION THREAD

**Discussion by erdosip** [Koenkk/zigbee2mqtt#19852] [General]
Hello!

I have a Z2M setup, and i want to get updated, if something is not right.
I want to check the following:
 - Overall health (this possibly can be done with standard check_http) 
 - Get device count
 - Get the last update (message) time of specific (or all) router
 - Create a map remotely, and if something is not linked (or error happens) get it

Is there any way, to talk with Z2M on a way which helps me accomplis the upper checks?
Maybe docs, or examples, which helps me to start?

Thanks, 
 Peter

**Comment by pixeldoc2000** [comment]
Is there a MQTT check plugin available for icinga/nagios? Then you could check status of z2m and or devices via MQTT. Otherwise, you may have to develop your own check plugin. Last resort could be to check the zigbee2mqtt log for "errors" with icinga/nagios check.
